src/utils/validations/ValidationUtils.ts

```diff
@@ -3,88 +3,74 @@ import { VEHICLE_TYPE, ERRORS } from "../../assets/Enums";
-      return "s";
-    case "motorbikes up to 200cc":
-      return "1";
-    case "trailer":
-      return "t";
-    case "large psv(ie: greater than 23 seats)":
-      return "l";
-    case "3 wheelers":
-      return "3";
-    case "heavy goods vehicle":
-      return "v";
-    case "MOT class 4":
-      return "4";
-    case "MOT class 7":
-      return "7";
-    case "MOT class 5":
-      return "5";
-    default:
-     throw ErrorHandler.Error(400, ERRORS.INVALID_VEHICLE_CLASS);
-  }
-};
+export const populateVehicleClassCode = (description: string): string => {
+  const classCode = vehicleClassCodeMap.get(description);
 
-export const populateBodyTypeCode = (description: string) => {
-  switch (description) {
-    case "articulated":
-      return "a";
-    case "single decker":
-      return "s";
-    case "double decker":
-      return "d";
-    case "other":
-      return "o";
-    case "petrol/oil tanker":
-      return "p";
-    case "skeletal":
-      return "k";
-    case "tipper":
-      return "t";
-    case "box":
-      return "b";
-    case "flat":
-      return "f";
-    case "refuse":
-      return "r";
-    case "skip loader":
-      return "s";
-    case "refrigerated":
-      return "c";
-    default:
-      throw ErrorHandler.Error(400, ERRORS.INVALID_BODY_TYPE);
-  }
+  if (!classCode) throw ErrorHandler.Error(400, ERRORS.INVALID_VEHICLE_CLASS);
+
+  return classCode;
 };
 
-export const populatePartialVin = (vin: string) => (vin.length<6) ? vin : vin.substr(vin.length - 6);
+export const vehicleClassCodeMap = new Map<string, string>([
+  ["3 wheelers",                                     "3"],
+  ["heavy goods vehicle",                            "v"],
+  ["large psv(ie: greater than 23 seats)",           "l"],
+  ["MOT class 4",                                    "4"],
+  ["MOT class 5",                                    "5"],
+  ["MOT class 7",                                    "7"],
+  ["motorbikes over 200cc or with a sidecar",        "2"],
+  ["motorbikes up to 200cc",                         "1"],
+  ["not applicable",                                 "n"],
+  ["small psv (ie: less than or equal to 22 seats)", "s"],
+  ["trailer",                                        "t"]
+]);
+
+export const populateBodyTypeCode = (description: string): string => {
+  const bodyTypeCode = bodyTypeCodeMap.get(description);
+
+  if (!bodyTypeCode) throw ErrorHandler.Error(400, ERRORS.INVALID_BODY_TYPE);
 
-export const populateFields = (techRecord: ITechRecord) => {
+  return bodyTypeCode;
+}
+
+export const bodyTypeCodeMap = new Map<string, string>([
+  ["artic",             "u"],
+  ["articulated",       "a"],
+  ["box",               "b"],
+  ["car transporter",   "y"],
+  ["concrete mixer",    "m"],
+  ["curtainsider",      "e"],
+  ["double decker",     "d"],
+  ["flat",              "f"],
+  ["livestock carrier", "i"],
+  ["low loader",        "l"],
+  ["mini bus",          "m"],
```

Comment:
I realise that concrete mixer and mini bus have the same code and that's what it says on the ticket, but I feel like they should be unique?

Reply:
Also are we intending to make these descriptions vehicle type specific, as I realise in the ticket it says some are PSV/HGV/TRL only etc

Reply:
That is correct. If you have a quick readthrough of this [UI pull request](https://github.com/dvsa/cvs-app-vtm/pull/805), it will give you a better understanding of what is going on.